sql/sqlwire/wire.proto

```diff
@@ -102,6 +100,16 @@ message Request {
   repeated Datum params = 3 [(gogoproto.nullable) = false];
 }
 
+// ResponseWithError is what we send on the wire. It includes the error so that
+// Response doesn't need to, simplifying error handling that involves handling
+// these objects.
+message ResponseWithError {
```

Comment:
I too like using the compiler to make changes like this. You could have accomplished the same by renaming `SetGoError` to something like `SetError` or moving it to the `sqlserver` package now that it only has a single use.